centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java

```diff
@@ -659,6 +673,10 @@ private void handleAsyncReply(Protocol.Reply reply) {
                     PublishEvent event = new PublishEvent();
                     event.setData(pub.getData().toByteArray());
                     sub.getListener().onPublish(sub, event);
+                }else if(this.serverSide.containsKey(channel)){
+                    MessageEvent event = new MessageEvent();
```

Comment:
Server side subs implementation is wrong -  we need new handlers on top level for server side publications. I suggest only add connect data in this pull request and continue work on server side subs in another pr. I can explain a bit more how this should work and why in Discord. What do you think?

Reply:
@FZambia i have removed server side subs implementation.  and formatter set connect data code.